cutils.h

```diff
@@ -356,6 +357,77 @@ static inline void inplace_bswap32(uint8_t *tab) {
     put_u32(tab, bswap32(get_u32(tab)));
 }
 
+static inline double fromfp16(uint16_t v) {
+    double d, s;
+    int e;
+    if ((v & 0x7C00) == 0x7C00) {
+        d = (v & 0x3FF) ? NAN : INFINITY;
+    } else {
+        d = (v & 0x3FF) / 1024.;
+        e = (v & 0x7C00) >> 10;
+        if (e == 0) {
+            e = -14;
+        } else {
+            d += 1;
+            e -= 15;
+        }
+        d = scalbn(d, e);
```

Comment:
If there's a way to implement fromfp16() and tofp16() without using functions from math.h, I'd love to hear it!

My initial approach used plain bit shifting but I couldn't get rounding to work correctly in all cases. Ex. test262 expects `Math.fp16round(32767)` to round up to 32,768.

For anyone interested, here's my first attempt:
<details>

```c
static inline double fromfp16(uint16_t v) {
    union {
        uint16_t v;
        struct {
            uint16_t f:10;
            uint16_t e:5;
            uint16_t s:1;
        } x;
    } h;
    union {
        double v;
        struct {
            uint64_t f:52;
            uint64_t e:11;
            uint64_t s:1;
        } x;
    } q;
    h.v = v;
    if (h.x.e == 31) { // +/-Infinity, NaN
        q.x.f = h.x.f > 0;
        q.x.e = 2047;
    } else if (h.x.e == 0 && h.x.f == 0) { // TODO subnormal if h.x.f > 0
        q.x.f = 0;
        q.x.e = 0;
    } else {
        q.x.f = (uint64_t)h.x.f << 42;
        q.x.e = (h.x.e - 15) + 1023;
    }
    q.x.s = h.x.s;
    return q.v;
}

static inline uint16_t tofp16(double d) {
    union {
        uint16_t v;
        struct {
            uint16_t f:10;
            uint16_t e:5;
            uint16_t s:1;
        } x;
    } h;
    union {
        double v;
        struct {
            uint64_t f:52;
            uint64_t e:11;
            uint64_t s:1;
        } x;
    } q;
    q.v = d;
    if (q.x.e == 2047)
        return 0x7C00 | (q.x.s << 15) | (q.x.f > 0); // +/-Infinity, NaN
    if (d < -65504 || d > 65504)
        return 0x7C00 | (q.x.s << 15); // out of range, return Infinity
    if (q.x.e == 0 && q.x.f == 0) {
        return 0;
    }
    h.x.f = q.x.f >> 42;
    h.x.e = (q.x.e - 1023) + 15;
    h.x.s = q.x.s;
    return h.v;
}
```

</details>

Reply:
If we really want to avoid the dependency on math.h for this very function perhaps we can borrow it from musl? https://git.musl-libc.org/cgit/musl/tree/src/math/scalbn.c

Reply:
Never mind, I see we use more math functions below.